libtelnet.c

```diff
@@ -325,17 +327,25 @@ static INLINE void _set_rfc1143(telnet_t *telnet, unsigned char telopt,
 	 * to the number of enabled options for most simple code, and it
 	 * allows for an acceptable number of reallocations for complex code.
 	 */
-	if ((qtmp = (telnet_rfc1143_t *)realloc(telnet->q,
-			sizeof(telnet_rfc1143_t) * (telnet->q_size + 4))) == 0) {
-		_error(telnet, __LINE__, __func__, TELNET_ENOMEM, 0,
-				"realloc() failed: %s", strerror(errno));
-		return;
+
+#define QUANTUM 4
```

Comment:
Can we rename this to something a bit more meaningful without the immediate context, and place it up with the other defines? Maybe something like `Q_BUFFER_GROWTH_QUANTUM` ?

The change's logic looks really good.